webhook/webhook.py

```diff
@@ -26,24 +30,69 @@
 app.config['DEBUG'] = True
 
 @app.route('/')
-def hello_world():
+def status():
     return 'Airflow 2.5.3 `git pull` webhook @ ' + time.ctime()
 
 @app.route('/webhook', methods=['POST'])
 def handle_webhook():
-    payload = request.get_json()
+    # we only want to run this in production, so we check the environment variable
+    if OPERATING_ENVIRONMENT != "production":
+        app.logger.info("\nnot in production, skipping webhook\n")
+        return "Not in production, skipping webhook"
+    # get the client key from the request header
+    client_key = request.headers.get('X-Webhook-Key')
+    # get the secret values from 1Password
     SECRET_VALUES = onepasswordconnectsdk.load_dict(client, SECRETS)
-
-    if payload["data"]["webhook_key"] == SECRET_VALUES["secret_value"]:
-        sys.stderr.write("successful webhook auth\n")
-        os.chdir('/opt/airflow')
-        environment = dict(os.environ)
-        environment['GIT_SSH_COMMAND'] = 'ssh -i /opt/private_key_for_github -o IdentitiesOnly=yes -o "StrictHostKeyChecking=no"'
-        data = subprocess.run(['git', 'pull'], env=environment, cwd=r'/opt/airflow', check=True, stdout=subprocess.PIPE).stdout
-        return 'Received webhook payload'
+    # check if the the access token matches
+    if client_key == SECRET_VALUES["webhook_token"]:
```

Comment:
i think this is pretty low-risk, but the recommended way to evaluate a password is to use the `hmac` library to avoid [a timing attack](https://sqreen.github.io/DevelopersSecurityBestPractices/timing-attack/python). we came across this somewhere in the Moped API but i can't find it at the moment.

Reply:
Huge TIL for me here. I strongly support this suggestion!

Reply:
one other thing: is there a chance the client key and secret both = `None`? i think you want to add an assertion that the `SECRET_VALUES["webhook_token"]` is not falsey.